src/js/player.js

```diff
@@ -1903,6 +1903,13 @@ class Player extends Component {
         // the tech loop to check for a compatible technology
         this.sourceList_([source]);
       } else {
+        let previous = this.currentSrc();
+
+        this.trigger('beforesourcechange', {
```

Comment:
My thought was that we'd only call it when we are changing to a playable source because that's what people will care about in most cases, I think.